src/Debug/skeletonViewer.ts

```diff
@@ -744,6 +746,10 @@ export class SkeletonViewer {
                 this.debugMesh.alwaysSelectAsActiveMesh = true;
             }
 
+            const light = this.utilityLayer!._getSharedGizmoLight();
+            light.includedOnlyMeshes = light.includedOnlyMeshes.concat(this.debugMesh!);

```

Comment:
This will add in `includedOnlyMeshes ` the (same) mesh each time `_buildSpheresAndSpurs` is called.

You should check if the mesh is not already in the array before adding it.

Also, why not simply using `light.includedOnlyMeshes.push(this.debugMesh!)`?

Reply:
Lol I totally missed it

Reply:
Is it the same mesh? Each time we call this function we dispose and rebuild a new mesh, so I'm not sure if checking for the mesh in the included mesh list is going to be a useful check...

We have a few preconditions however:
1) each SkeletonViewer has its own unique UtilityLayer scene and light
2) each time we recreate the debug mesh we dispose the old one.

It might be best to even just set the light we use to just light the newly created debug mesh?

Reply:
I'll just remove this line, I don't think we're getting much out of it...